### DISCUSSION TITLE
Multiple sites in one setup

### DISCUSSION THREAD

**Discussion by Diewy** [aerni/statamic-advanced-seo#106] [Ideas]
Our Statamic projects have multiple sites with multipe languages:

- Main site - English
- Main site - French
- B2B site - English
- B2B site - French

But, when using the SEO plugin (and all other SEO plugins for Statamic) you'll see all the sites, not only the related ones.

For example:

![CleanShot 2023-11-08 at 11 02 45](https://github.com/aerni/statamic-advanced-seo/assets/1014380/d822bf38-79ee-475b-aa9e-3ac4a69884ff)

The same goes for `alternate` links.

Besides the head tags, the B2B website shouldn't be indexed. So it should be excluded in the Sitemap + a `noindex` should be added. This is already possible via the SEO settings. 👍 

I'm not sure what the best solution would be for the head tags. A site filter on the Antler variables?

**Comment by aerni** [comment]
I see your point. While you can use Statamic's multi-site feature to create multiple unrelated sites, it was mainly built around the concept of localizing the same site into different languages. There's a fundamental flaw that all sites are basically treated as translations of the same site.

To make it work the way you're describing, we'd have to be able to "group" sites to effectively distinguish between an actual "site" and the "localizations" of a site. This is a fundamental Statamic core issue and is a [feature request](https://github.com/statamic/ideas/issues/17) that has been raised a lot.

That being said, we might be able to hook into [additional site attributes](https://statamic.dev/multi-site#additional-attributes) to "group" sites and use that for Advanced SEO. But I rather just wait for this to be solved in core before investing a lot of time and resources to hack around Statamic's limitations.

**Comment by aerni** [comment]
This request will be implemented in v3 of this addon.

Sitemaps are now scoped by domain. Previously, sitemaps included URLs from all sites regardless of domain. Now, only URLs from sites sharing the same domain are included. This is a bug fix for multi-domain setups where URLs were incorrectly mixed.

| Setup | Behavior |
|-------|----------|
| `example.com`, `example.com/de`, `example.com/fr` | Unified sitemap (unchanged) |
| `example.com`, `example.de`, `example.fr` | Separate sitemaps per domain (bug fix) |

> **Note:** Hreflang tags are unaffected and continue to reference all localized versions across domains.
